meilisearch-types/src/document_formats.rs

```diff
@@ -111,17 +114,49 @@ pub fn read_csv(input: impl Read, writer: impl Write + Seek) -> Result<usize> {
     Ok(count as usize)
 }
 
-/// Reads JSON Lines from input and write an obkv batch to writer.
-pub fn read_ndjson(input: impl Read, writer: impl Write + Seek) -> Result<usize> {
+/// Reads JSON from temporary file  and write an obkv batch to writer.
+pub fn read_json(file: &File, writer: impl Write + Seek) -> Result<usize> {
+    read_json_inner(file, writer, PayloadType::Json)
+}
+
+/// Reads JSON from temporary file  and write an obkv batch to writer.
+pub fn read_ndjson(file: &File, writer: impl Write + Seek) -> Result<usize> {
+    read_json_inner(file, writer, PayloadType::Ndjson)
+}
+
+/// Reads JSON from temporary file  and write an obkv batch to writer.
+fn read_json_inner(
+    file: &File,
+    writer: impl Write + Seek,
+    payload_type: PayloadType,
+) -> Result<usize> {
     let mut builder = DocumentsBatchBuilder::new(writer);
-    let reader = BufReader::new(input);
-
-    for result in serde_json::Deserializer::from_reader(reader).into_iter() {
-        let object = result.map_err(Error::Json).map_err(|e| (PayloadType::Ndjson, e))?;
-        builder
-            .append_json_object(&object)
-            .map_err(Into::into)
-            .map_err(DocumentFormatError::Internal)?;
+    let mmap = unsafe { MmapOptions::new().map(file).unwrap() };
+    let mut deserializer = serde_json::Deserializer::from_slice(&mmap);
+
+    match array_each(&mut deserializer, |obj: Object| builder.append_json_object(&obj)) {
+        Ok(Ok(count)) => debug!("serde json array size: {}", count),
+        Ok(Err(e)) => return Err(DocumentFormatError::Internal(Box::new(e))),
+        Err(_e) => {
+            debug!("deserialize single json");
```

Comment:
Would it be possible to replace this debug log by a little more explicit comment, here?

Reply:
I'll adjust it.